lib/src/null_safety_analysis.dart

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:analyzer/dart/analysis/context_builder.dart';
+import 'package:analyzer/dart/analysis/context_locator.dart';
+import 'package:pub/src/source/path.dart';
+
+import 'package:pub_semver/pub_semver.dart';
+import 'package:path/path.dart' as path;
+
+import 'entrypoint.dart';
+import 'io.dart';
+import 'package_name.dart';
+import 'pubspec.dart';
+import 'solver.dart';
+import 'source/cached.dart';
+import 'system_cache.dart';
+
+enum NullSafetyCompliance {
+  /// This package and all dependencies opted into null safety.
+  compliant,
+
+  /// This package opted into null safety, but some file or dependency is not
+  /// opted in.
+  apiOnly,
+
+  /// This package did not opt-in to null safety yet.
+  notCompliant,
+
+  /// The resolution failed. Or some dart file in a dependency
+  /// doesn't parse.
+  analysisFailed,
+}
+
+class NullSafetyAnalysis {
+  final SystemCache _systemCache;
+  final Map<PackageId, Future<NullSafetyCompliance>>
+      _packageInternallyGoodCache = {};
+  static final _firstVersionWithNullSafety = Version.parse('2.10.0');
+
+  NullSafetyAnalysis(SystemCache systemCache) : _systemCache = systemCache;
+
+  Future<NullSafetyCompliance> nullSafetyCompliance(PackageId packageId) async {
+    final result = await _nullSafetyCompliance(packageId);
+    return result;
+  }
+
+  /// Returns true if package version [packageId] and all its non-dev
+  /// dependencies (transitively) have a language version >= 2.10, and no files
+  /// in lib/ of  these packages opt out to a pre-2.10 language version.
+  ///
+  /// This will do a full resolution of that package's import graph, and also
+  /// download the package and all dependencies into [cache].
+  Future<NullSafetyCompliance> _nullSafetyCompliance(
+      PackageId packageId) async {
+    return await withTempDir((tempPath) async {
+      final importingPubspec = {
+        'name': '${packageId.name}_importer',
```

Comment:
If we use a space as separator like we do in global activate

Reply:
Done